pkgs/development/compilers/mint/default.nix

```diff
@@ -0,0 +1,60 @@
+{stdenv, lib, fetchgit, crystal, zlib, openssl, duktape}:
+let
+  crystalPackages = lib.mapAttrs (name: src:
+    stdenv.mkDerivation {
+      name = lib.replaceStrings ["/"] ["-"] name;
+      src = fetchgit src;
+      phases = "installPhase";
+      installPhase = ''cp -r $src $out'';
+      passthru = { libName = name; };
+    }
+  ) (import ./shards.nix);
+
+  crystalLib = stdenv.mkDerivation {
+    name = "crystal-lib";
+    src = lib.attrValues crystalPackages;
+    libNames = lib.mapAttrsToList (k: v: [k v]) crystalPackages;
+    phases = "buildPhase";
+    buildPhase = ''
+      mkdir -p $out
+      linkup () {
+        while [ "$#" -gt 0 ]; do
+          ln -s $2 $out/$1
+          shift; shift
+        done
+      }
+      linkup $libNames
+    '';
+  };
+in
+stdenv.mkDerivation rec {
+  version = "2018-05-27";
+  name = "mint-${version}";
+  src = fetchgit {
+    url = "https://github.com/mint-lang/mint.git";
+    rev = "a3f0c86f54b8b3a18dda5c39c2089bdb1d774b4f";
+    sha256 = "1bgs6jkwfc2ksq4gj55cl3h2l5g25f5bwlsjryiw9cbx5k4bp1kz";
+  };
+
+  buildInputs = [ crystal zlib openssl duktape ];
+
+  buildPhase = ''
+    mkdir -p $out/bin
+
+    mkdir tmp
+    cd tmp
+    ln -s ${crystalLib} lib
+    cp -r $src/* .
+    crystal build src/mint.cr -o $out/bin/mint --verbose --progress --release --no-debug
+  '';
+
+  installPhase = ''true'';
+
+  meta = {
+    description = "A refreshing language for the front-end web.";
+    homepage = https://mint-lang.com/;
+    license = stdenv.lib.licenses.bsd3;
+    maintainers = with stdenv.lib.maintainers; [ manveru ];
+    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" ];
```

Comment:
Why just these platforms? IIRC crystal supports others too, e.g. aarch64.

Reply:
Crystal only has releases for these platforms: https://github.com/crystal-lang/crystal/releases/tag/0.24.2 and since you need Crystal to build Crystal, i don't think we'll have other platforms soon. Please correct me if I'm wrong, but those platforms are also set in the meta of the crystal package itself.